packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -357,6 +358,10 @@ export interface GridOptions {
    * Set the density of the grid.
    */
   density: Density;
+  /**
+   * Set the locale text of the grid.
+   */
+  localeText: object;
```

Comment:
How about we type a bit better?
`{[key: string]: string}` or `Record<string, string>`

Reply:
done

Reply:
We will also need to support react node

Reply:
I've created a custom type to cover the 3 different cases -> `string` | `React.ReactNode` | `Function`. The only inconvenience with this is that depending on the types  we may need to cast it depending on the place where it is used, like in here `label: apiRef!.current.getText('densityOptionCompact') as string`. I really don't want to change the typings of existing interfaces and also that is more a corner case. In the majority of the time plain text is used directly, so it wouldn't be a problem.

Reply:
@DanailH What do you think about listing all the keys?

Reply:
I would expect most of the keys to be react node, sometime we might only support a string when it needs to be applied on an HTML attribute. We might also need a function, from time to time. You can check the type of the translated strings in the existing components of the main repository to get an idea on the frequency of each case.

Reply:
We can definitely list all the keys but this list will grow over time. The approach of generalising looked simpler at the time. I'll check the typings in the main repo as well.

Reply:
I think that from a DX perspective, listing all the keys will be better. The overhead for us seems limited.

Reply:
I think we cleared out the typings - I've updated the code with the new typings. I'll resolve this and if need to discuss it again we can open a new discussion thread.